app/User.php

```diff
@@ -208,4 +209,26 @@ public function fullName()
         }
         return "";
     }
+
+    /**
+     * Ecrypt the user's google_2fa secret.
+     *
+     * @param  string  $value
+     * @return string
+     */
+    public function setGoogle2faSecretAttribute($value)
+    {
+        $this->attributes['google2fa_secret'] = encrypt($value);
+    }
+
+    /**
+     * Decrypt the user's google_2fa secret.
+     *
+     * @param  string  $value
+     * @return string
+     */
+    public function getGoogle2faSecretAttribute($value)
+    {
+        return decrypt($value);
```

Comment:
Following https://laravel.com/docs/5.7/encryption#using-the-encrypter, I think this should be wrapped in a try / catch

Reply:
hmm, okay, but what to do when this happens?

Reply:
In this case maybe invalidate the secret and prompt the user to re establish 2fa? I'm not quite sure when this can happen ...

Reply:
if this fails to decrypt, it means that the main secret is broken
this should'nt happen and can only be fixed with help of an admin user who can reset the token
if you allow the user to set a new token by himself in this case you cannot garantee that the person who does this is the real person.
this means it can possibly used in an attak scenario.
thats the reason  why i would stay the code as it is.